src/app/modules/main/profile_section/ens_usernames/module.nim

```diff
@@ -0,0 +1,242 @@
+import NimQml, json, stint, strutils, strformat, parseutils, chronicles
+
+import io_interface
+import ../io_interface as delegate_interface
+import view, controller, model
+
+import ../../../../core/eventemitter
+import ../../../../../app_service/common/conversion as service_conversion
+import ../../../../../app_service/service/settings/service_interface as settings_service
+import ../../../../../app_service/service/ens/service as ens_service
+import ../../../../../app_service/service/ens/utils as ens_utils
+import ../../../../../app_service/service/wallet_account/service_interface as wallet_account_service
+
+export io_interface
+
+logScope:
+  topics = "profile-section-ens-usernames-module"
+
+include ../../../../../app_service/common/json_utils
+
+type 
+  Module* = ref object of io_interface.AccessInterface
+    delegate: delegate_interface.AccessInterface
+    view: View
+    viewVariant: QVariant
+    controller: controller.AccessInterface
+    moduleLoaded: bool
+
+proc newModule*(delegate: delegate_interface.AccessInterface, events: EventEmitter, 
+  settingsService: settings_service.ServiceInterface, ensService: ens_service.Service,
+  walletAccountService: wallet_account_service.ServiceInterface): Module =
+  result = Module()
+  result.delegate = delegate
+  result.view = view.newView(result)
+  result.viewVariant = newQVariant(result.view)
+  result.controller = controller.newController(result, events, settingsService, ensService, walletAccountService)
+  result.moduleLoaded = false
+  
+method delete*(self: Module) =
+  self.view.delete
+  self.viewVariant.delete
+  self.controller.delete
+
+method load*(self: Module) =
+  self.controller.init()
+  
+  let signingPhrase = self.controller.getSigningPhrase()
+  let network = self.controller.getCurrentNetworkDetails()
+  var link = network.etherscanLink.replace("/address", "/tx")
+  self.view.load(link, signingPhrase)
+
+method isLoaded*(self: Module): bool =
+  return self.moduleLoaded
+
+method viewDidLoad*(self: Module) =
+  self.fetchGasPrice()
+  # add registered ens usernames
+  let registeredEnsUsernames = self.controller.getAllMyEnsUsernames(false)
```

Comment:
Can you put the param name for clarity in from of `false`

Reply:
@jrainville updated.